Apps/Src/CAN_Queue.c

```diff
@@ -10,39 +12,106 @@
 
 // fifo
 #define FIFO_TYPE CANMSG_t
-#define FIFO_SIZE 256
-#define FIFO_NAME CAN_fifo
+#define FIFO_SIZE 512
+#define FIFO_NAME CAN_fifo_TRANSMIT
 #include "fifo.h"
 
-static CAN_fifo_t canFifo;
-static OS_SEM canFifo_Sem4;
-static OS_MUTEX canFifo_Mutex;
+#define FIFO_TYPE CANMSG_t
+#define FIFO_SIZE 512
+#define FIFO_NAME CAN_fifo_RECEIVE
+#include "fifo.h"
+
+static CAN_fifo_TRANSMIT_t canFifo_TRANSMIT;
+static CAN_fifo_RECEIVE_t canFifo_RECEIVE;
+
+static OS_SEM canFifo_Transmit_Sem4;
+static OS_MUTEX canFifo_Transmit_Mutex;
+
+static OS_SEM canFifo_Receive_Sem4;
+static OS_MUTEX canFifo_Receive_Mutex;
 
+
+/**
+ * @brief: initializes both Receive and Transmit fifo + sema4s and mutexes
+ * @param: none
+ * @return: none
+*/
 void CAN_Queue_Init(void) {
-    RTOS_BPS_MutexCreate(&canFifo_Mutex, "CAN queue mutex");
-    RTOS_BPS_SemCreate(&canFifo_Sem4, "CAN queue semaphore", 0);
-    RTOS_BPS_MutexPend(&canFifo_Mutex, OS_OPT_PEND_BLOCKING);
-    CAN_fifo_renew(&canFifo);
-    RTOS_BPS_MutexPost(&canFifo_Mutex, OS_OPT_POST_NONE);
+    RTOS_BPS_MutexCreate(&canFifo_Transmit_Mutex, "CAN transmit queue mutex");
+    RTOS_BPS_SemCreate(&canFifo_Transmit_Sem4, "CAN transmit queue semaphore", 0);
+    
+    RTOS_BPS_MutexCreate(&canFifo_Receive_Mutex, "CAN receive queue mutex");
+    RTOS_BPS_SemCreate(&canFifo_Receive_Sem4, "CAN receive queue semaphore", 0);
+    
+    //TODO: i dont think these mutexes are necessary honestly, maybe remove??
```

Comment:
yuh im pretty sure u want teh mutexes

Reply:
you shouldn't need it cause you should never initialize can queue twice. doesn't hurt i guess